lib/tools/system-calls.js

```diff
@@ -558,7 +559,13 @@ systemCallMethods.reboot = async function (retries = DEFAULT_ADB_REBOOT_RETRIES)
 
 systemCallMethods.root = async function () {
   try {
-    await exec(this.executable.path, ['root']);
+    let {stdout} = await exec(this.executable.path, ['root']);
+
+    // on real devices in some situations we get an error in the stdout
+    if (stdout.indexOf('adbd cannot run as root') !== -1) {
```

Comment:
strange that exec does not return non-zero return code (and throws an exception) in such case